syntax = "proto3";

package tag_suggestion.v1;

service TagSuggestionService {
  rpc Suggest(SuggestRequest) returns (SuggestResponse);
}

message SuggestRequest {
  repeated string image_urls = 1;
}

message SuggestionScore {
    uint64 tag_id = 1;

  // The score of the suggestion. Higher scores are better.
    double score = 2;
}

message Suggestion {
  string image_url = 1;
  // The index is matching with a tag id
  // scores is sorted by tags with higher scores
  repeated SuggestionScore scores = 2;
}

message Tag {
    uint64 id = 1;
    string name = 2;
}

message SuggestResponse {
  repeated Suggestion suggestions = 1;

  // The all tags that can be potentially suggested
  // The key is the tag id
  map<uint64, Tag> all_tags = 2;
}
